src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py

```diff
@@ -268,35 +295,40 @@ def up(self) -> None:
             return
 
         logger.info("Spinning up local Kubeflow Pipelines deployment...")
-        fileio.make_dirs(self.root_directory)
-        container_registry_port = int(container_registry.uri.split(":")[-1])
-        container_registry_name = self._get_k3d_registry_name(
-            port=container_registry_port
-        )
-        local_deployment_utils.write_local_registry_yaml(
-            yaml_path=self._k3d_registry_config_path,
-            registry_name=container_registry_name,
-            registry_uri=container_registry.uri,
-        )
-        local_deployment_utils.create_k3d_cluster(
-            cluster_name=self._k3d_cluster_name,
-            registry_name=container_registry_name,
-            registry_config_path=self._k3d_registry_config_path,
-        )
-        kubernetes_context = f"k3d-{self._k3d_cluster_name}"
-        local_deployment_utils.deploy_kubeflow_pipelines(
-            kubernetes_context=kubernetes_context
-        )
-        local_deployment_utils.start_kfp_ui_daemon(
-            pid_file_path=self._pid_file_path,
-            port=self.kubeflow_pipelines_ui_port,
-        )
 
-        logger.info(
-            f"Finished local Kubeflow Pipelines deployment. The UI should now "
-            f"be accessible at "
-            f"http://localhost:{self.kubeflow_pipelines_ui_port}/."
-        )
+        try:
+            fileio.make_dirs(self.root_directory)
+            container_registry_port = int(container_registry.uri.split(":")[-1])
+            container_registry_name = self._get_k3d_registry_name(
+                port=container_registry_port
+            )
+            local_deployment_utils.write_local_registry_yaml(
+                yaml_path=self._k3d_registry_config_path,
+                registry_name=container_registry_name,
+                registry_uri=container_registry.uri,
+            )
+            local_deployment_utils.create_k3d_cluster(
+                cluster_name=self._k3d_cluster_name,
+                registry_name=container_registry_name,
+                registry_config_path=self._k3d_registry_config_path,
+            )
+            kubernetes_context = f"k3d-{self._k3d_cluster_name}"
+            local_deployment_utils.deploy_kubeflow_pipelines(
+                kubernetes_context=kubernetes_context
+            )
+            local_deployment_utils.start_kfp_ui_daemon(
+                pid_file_path=self._pid_file_path,
+                port=self.kubeflow_pipelines_ui_port,
+            )
+            logger.info(
+                f"Finished local Kubeflow Pipelines deployment. The UI should now "
+                f"be accessible at "
+                f"http://localhost:{self.kubeflow_pipelines_ui_port}/."
+                f"The orchestrator is now up."
+            )
+        except Exception as e:
+            self.list_manual_setup_steps()
+            raise e
```

Comment:
Not sure how to handle it exactly, but I assume raising the exception here will probably hide the steps to run it manually for most users?

Reply:
@schustmi So you think we should just handle the exception and log our own error messages? Might be annoying if someone wants to know what went wrong.

Reply:
Also I'm not sure I understand why the manual setup steps would be hidden.

Reply:
By hidden I mean the exception log is usually quite long and the user might not see our previous logs because they might have scrolled out of bounds of the terminal window. But I definitely agree that we should still show the error message somehow, maybe we can log the exception before our manual explanations?

Reply:
Agreed. I did it this way. Once we've added `rich` into our arsenal that might help with the readability, I think.